apps/nouns-camp/src/hooks/simulation.js

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import { encodeFunctionData } from "viem";
+
+const SIMULATE_TRANSACTION_TYPES = [
+  "function-call",
+  "proxied-function-call",
+  "payable-function-call",
+  "proxied-payable-function-call",
+  "treasury-noun-transfer",
+  "escrow-noun-transfer",
+  "transfer",
+  "usdc-transfer-via-payer",
+];
+
+export const fetchSimulation = async (transaction) => {
```

Comment:
I think you’d be better off "unparsing" the transaction before we try to simulate to more closely mirror what will happen when submitting the proposal (and hopefully also catch more errors). That way you also wouldn’t have to care about the messy camp tx format.

e.g.

```js
// Unparsing gives you raw targets, signatures, calldatas, and values
fetchSimulation(unparse([transaction]))
```

Then decode the call inputs again with viem, something like:

```js
    const { name, inputs } = parseAbiItem(`function ${signature}`);
    const args = decodeAbiParameters(inputs, calldata);
```